src/krate.rs

```diff
@@ -1236,6 +1262,13 @@ pub fn new(req: &mut Request) -> CargoResult<Response> {
     })
 }
 
+/// Used by `krate::new` function.
+///
+/// This function parses the JSON headers for validity before returning the
```

Comment:
I wouldn't say it parses the JSON *for* validity, it parses the JSON so that we can interpret the data, and during and after parsing it is also checking the validity of the data :)